tools/kmod/sof_remove.sh

```diff
@@ -5,8 +5,17 @@
 
 TOPDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)
 
-# shellcheck source=case-lib/lib.sh
-source "$TOPDIR"/case-lib/lib.sh
+# Used only by more advanced error handling
+source_opt_libsh()
+{
+    local libsh="$TOPDIR"/case-lib/lib.sh
+    if test -e "$libsh"; then
+        # shellcheck source=case-lib/lib.sh
+        source "$libsh"
+    else
+        printf '%s warning: %s not found, not sourced\n' "$0" "$libsh"
+    fi
+}
```

Comment:
I did understand.

This is software: it can be BOTH dumb and smart and this is what this PR does. Try it?

Reply:
The moved `sof_remove.sh` do work now.
Apparently it also works in place as well, not failing anymore if the device were booted in legacy or DSPless mode..

Reply:
> Apparently it also works in place as well, not failing anymore if the device were booted in legacy or DSPless mode..

Probably because of what I did it a while back in commit 7c2ede03c9be , PR #1187 and friends.

(this was much more work than this)